lib/dao.js

```diff
@@ -625,7 +625,172 @@ DataAccessObject.upsert = function(data, options, cb) {
   }
   return cb.promise;
 };
+/**
+ * Update or insert a model instance based on the search criteria.
+ * If there is a single instance retrieved,update the model instance.
+ * If there are no model instances retrieved then create a new model instance and
+ * If there are multiple instances retreived then upsertWithWhere operation will return error.
```

Comment:
`retreived` spelling typo, also you can change the sentence to

Returns an error if multiple instances are retrieved.